### ISSUE TITLE
JavaScript struct field tags don't print correctly (due to reflection) unless accessed directly.

### ISSUE THREAD

**Issue by shurcooL** [gopherjs/gopherjs#141]
**Summary:** For example `fmt.Println(s)` produces unexpected results versus `fmt.Println(s.Foo)`.

This may or may not be a duplicate of #79 or #81, I couldn't really determine what specifically those issues are about and what remains for them to be closed.

However, now that `js:"someField"` struct field tags are supported, there is an issue where directly accessing the fields works as expected, but accessing indirectly (via reflection?) does not.

```Go
package main

import (
	"fmt"
	"honnef.co/go/js/dom"
	"github.com/gopherjs/gopherjs/js"
)

// type KeyboardEvent struct {
//     ...
//     KeyCode       int    `js:"keyCode"`

func main() {
	dom.GetWindow().Document().(dom.HTMLDocument).Body().AddEventListener("keydown", false, func(event dom.Event) {
		ke := event.(*dom.KeyboardEvent)

		fmt.Println("ke.KeyCode:", ke.KeyCode)          // Prints 71.
		fmt.Printf("ke: %#v\n", ke)                     // Prints all zeros.
		js.Global.Get("console").Call("log", ke.Object) // Prints KeyCode: 71 and other actual non-zero values of js.Object.
	})
}
```

![image](https://cloud.githubusercontent.com/assets/1924134/5560402/137ff64a-8d3b-11e4-847e-f3e15afcfe9f.png)

`fmt.Printf("ke: %#v\n", ke)` uses reflection rather than accessing struct fields directly, and GopherJS does not rewrite/substitute with the `js.Object` values instead.

This is unexpected and should be documented (it may already be, sorry if I missed it) or fixed if possible.

**Comment by neelance** [comment]
I know how to fix this issue, I just have to find some time to do it.

**Comment by shurcooL** [comment]
Confirmed fixed in e8b22e309cbbfd73eeae8c75dac06bb00d9863e1.

Before:

![image](https://cloud.githubusercontent.com/assets/1924134/6540786/2d37ebea-c465-11e4-8c67-fe7c2b4e9ae5.png)

After:

![image](https://cloud.githubusercontent.com/assets/1924134/6540787/36cf90ae-c465-11e4-9f02-c05b6a294d37.png)
